### DISCUSSION TITLE
Suggestions for regions

### DISCUSSION THREAD

**Discussion by ETCore7** [Acly/krita-ai-diffusion#1475] [Ideas]
Edit.

- When working with regions, would it be possible to have a toggle to always see the end result in Live mode, instead of raw layers with a dark raster on top? Changes to any region do affect other regions, but it's invisible now until you switch to a layer/region that covers the entire canvas. An isolated region may render something totally different when no other regions are nearby.

- When fetching the end result ('Create a new layer with the current result') original regions get messed up. They lose their local prompt and they are put in groups. The new layer with the end result can be somewhere in a group, or it is scattered over local groups. Is it possible to just get the end result as a new layer on top of everything else, while layers/regions do not get grouped nor have their local prompts deleted (nor that layers get renamed)?

- When creating a new layer while you're in a layer with a region the interface will show as if the same region is also associated with the new layer. Only after switching to another layer and back the interface correctly shows no region has been assigned yet.

As always thank you.

**Comment by Sorry-Im-Stupid** [comment]
I can't figure out how to even get it to generate something meaningful in a region. I'm trying to make a comic page, so I've sectioned off into a simple three panel cutout to test, and when I select one of the panels and generate, it just makes a weird blobby mess.
![why mess](https://github.com/user-attachments/assets/098ea287-b10d-4df2-91a9-a65ee4e594df)

This happens no matter what kind of fill I try, except for Replace background, which does at least generate something, but it's still a mess, and it completely removes the lines.
![replace what](https://github.com/user-attachments/assets/021c4f40-a73d-444b-85da-057c179b7494)

**Reply by dauwpe** [reply]
I think you have to fill the region with color, and it seems like you are describing the general prompt there? For me regions do work decently, although they have the habit to also use the prompts from underlying regions.

**Comment by ETCore7** [comment]
It would be nice if the plugin could act like a layer in Krita so the end result is AI generated stuff with a hard baked overlay/mask with fixed things on top (like a cartoon cutout template), but that would probably require a lot of work.

I can't see whether you're in live mode or not, the screenshots don't open to 100% here. You could try to assign a line art controlnet to the cutout and set it strong, but keep in mind that the AI always hallucinates a complete image. It does not construct lose objects and merge baked components.

As for regions, there must be something in the layers at the location where you want a local region prompt to render something. The plugin will try to follow the shape, but it's not 100%. You can ie draw a crude toy car in an empty layer and assign a region with local prompt 'toy car', and move it around, but the plugin may render it as a car on a poster or as something completely different when it is not on a logical place, like on a desk.

**Comment by Acly** [comment]
Regions in live mode are meant to follow the layer/region you're currently working on. They only show part of the image, but also only modify part of the image. It's true that because of "soft edge" results will "splat" to neighbouring regions a bit, but only at the edges, and not beyond what is shown in the preview.

Generating the entire image with all region information is something different, and doesn't happen in live mode.

> When fetching the end result ('Create a new layer with the current result') original regions get messed up.

Do you have a .kra file with a setup just _before_ you apply and create the new layers? That would be helpful to reproduce. Usually converting layers into groups should preserve the links.

> When creating a new layer while you're in a layer with a region the interface will show as if the same region is also associated with the new layer.

I think I know why, can probably fix that.
